### ISSUE TITLE
Asynchronous processing does not work when `TomcatService` and `Spring` are combined.

### ISSUE THREAD

**Issue by heowc** [line/armeria#3737]
While solving the following issue, I found another issue. (Or maybe it's not an issue 😅)

[According to the Spring documentation](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html#mvc-ann-async), asynchronous processing can be performed by returning a specific type from the method of the Controller. However, when I test with armeria + tomcat + spring combination, I get an empty response. This doesn't solve the problem I'm trying to solve, and I don't know exactly where the problem is. (tomcat? spring?)

```java
@RestController
@RequestMapping("/greeting")
public class GreetingController {
    private static final String template = "Hello, %s!";

    @GetMapping("/async")
    public Callable<ResponseEntity<Greeting>> greetingAsync(
            @RequestParam(value = "name", defaultValue = "World") String name) {
        return () -> {
            final Greeting body = new Greeting(String.format(template, name));
            return ResponseEntity.ok(body);
        };
    }
}
```

```java
@Test
public void async() throws Exception {
    final ResponseEntity<String> entity = restTemplate.getForEntity("http://localhost:" +
                                                                    httpPort +
                                                                    "/tomcat/api/rest/v1/greeting/async",
                                                                    String.class);
    System.out.println(entity); // <-- <200,[content-length:"0", server:"Armeria/1.9.1-SNAPSHOT", date:"Mon, 2 Aug 2021 03:50:58 GMT"]>
    assertThat(entity.getBody())
            .contains("Hello, World!");
}
```

**Comment by minwoox** [comment]
Did you use `TomcatService`? IIRC, we didn't implement async support in `TomcatService` unlike `JettyService`.

**Comment by heowc** [comment]
> Did you use TomcatService? 

Yes, that's right.

Aha... Then I will close the issue 😅

**Comment by trustin** [comment]
It's a known issue that needs a fix - #116
